internal/devserver/main/main.go

```diff
@@ -1,28 +1,37 @@
 package main
 
 import (
+	"bufio"
 	"flag"
 	"fmt"
+	"io"
+	"io/ioutil"
 	"net/http"
 	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
 	"strings"
 
 	"github.com/bazelbuild/rules_typescript/internal/concatjs/concatjs"
 	"github.com/bazelbuild/rules_typescript/internal/devserver/devserver"
 )
 
 var (
-	port        = flag.Int("port", 5432, "server port to listen on")
-	base        = flag.String("base", "", "server base (required, runfiles of the binary)")
-	pkgs        = flag.String("packages", "", "root package(s) to serve, comma-separated")
-	manifest    = flag.String("manifest", "", "sources manifest (.MF)")
-	servingPath = flag.String("serving_path", "/_/ts_scripts.js", "path to serve the combined sources at")
+	port             = flag.Int("port", 5432, "server port to listen on")
+	base             = flag.String("base", "", "server base (required, runfiles of the binary)")
+	pkgs             = flag.String("packages", "", "root package(s) to serve, comma-separated")
+	manifest         = flag.String("manifest", "", "sources manifest (.MF)")
+	scriptsManifest  = flag.String("scripts_manifest", "", "preScripts manifest (.MF)")
+	livereloadScript = flag.String("livereload_script", "", "path to live reload script")
```

Comment:
Alternative design: `IBAZEL_LIVERELOAD_SCRIPT` contains the full URL to the host serving the `.js` file that has the livereload implementation. The script can be fully self contained, including the right configuration (that you parse from the script below).

`concatjs` then just adds an IIFE loading the script via a `<script>` tag:

```javascript
(function() {
  var scriptTag = document.createElement('script');
  scriptTag.src = '${IBLAZE_LIVERELOAD_URL}';
  document.body.appendChild(scriptTag);  // or, a bit nicer, find the last `<script>` tag and append there.
}());
```

Assuming the script at the given URL then configures livereload properly (I think it even defaults to use the host it was served from, so could work with no further config), that saves a bunch of code here.

Reply:
Thanks. I think you were looking at an old version as I realized the same thing and changed it on Sat morning.